provider/server/server_suite_test.go

```diff
@@ -61,24 +65,26 @@ var _ = BeforeSuite(func() {
 				MemoryBytes: 8589934592,
 			},
 		},
+		{
+			Name: machineClassx2medium,
+			Capabilities: &iriv1alpha1.MachineClassCapabilities{
+				CpuMillis:   2000,
+				MemoryBytes: 2147483648,
+			},
+		},
 	}
 	machineClassData, err := json.Marshal(machineClasses)
 	Expect(err).NotTo(HaveOccurred())
-
-	machineClassesFile, err := os.CreateTemp(GinkgoT().TempDir(), "machineclasses")
+	machineClassesFile, err = os.CreateTemp(GinkgoT().TempDir(), "machineclasses")
 	Expect(err).NotTo(HaveOccurred())
-	defer func() {
-		_ = machineClassesFile.Close()
-	}()
 	Expect(os.WriteFile(machineClassesFile.Name(), machineClassData, 0600)).To(Succeed())
+	DeferCleanup(machineClassesFile.Close)
+	DeferCleanup(func() error {
+		return os.Remove(machineClassesFile.Name())
```

Comment:
What are you doing with error returned into DeferCleanup, i don't think it will check automatically.

Reply:
Added error check in DeferCleanup